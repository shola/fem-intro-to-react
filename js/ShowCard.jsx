// @flow

import React from 'react';
import { ShowPropsType, ShowStateType } from './ShowTypes';

/**
 * We must add keys to components that will need to render in different places
 * in the DOM. React knows how to re-use keyed components, instead of 
 * creating new ones and wasting resources.
 */
const ShowCard = (props: ShowPropsType): React.Element<any> => {
    const {
        title,
        imdbID,
        poster,
        year,
        description
    } = props;
    return (
        <div className="show-card" key={imdbID}>
            <img className="show-card-img" src={`public/img/posters/${poster}`} />
            <div className="show-card-text">
                <h3 className="show-card-title">{title}</h3>
                <h4 className="show-card-year">({year})</h4>
                <p className="show-card-description">{description}</p>
            </div>
        </div>
    );
};

export default ShowCard;
