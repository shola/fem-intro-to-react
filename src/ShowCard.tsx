import * as React from 'react';

interface ShowPropsType {
    title: string,
    imdbID: string,
    poster: string,
    year: string,
    description: string,
    trailer: string
}

interface ShowStateType {
    searchTerm: string
}

const ShowCard = (props: ShowPropsType) => {
    const {
        title,
        imdbID,
        poster,
        year,
        description
    } = props;
    return (
        <div className="show-card" key={imdbID}>
            <img className="show-card-img" src={`public/img/posters/${poster}`} />
            <div className="show-card-text">
                <h3 className="show-card-title">{title}</h3>
                <h4 className="show-card-year">({year})</h4>
                <p className="show-card-description">{description}</p>
            </div>
        </div>
    );
};

export default ShowCard;
