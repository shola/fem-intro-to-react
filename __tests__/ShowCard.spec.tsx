import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as Renderer from 'react-test-renderer';
import ShowCard from '../src/ShowCard';

xdescribe('ShowCard View', () => {
    it('Check snapshot for unexpected changes', () => {
        const component = Renderer.create(<ShowCard />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
