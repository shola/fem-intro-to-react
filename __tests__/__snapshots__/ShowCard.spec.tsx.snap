// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowCard View Check snapshot for unexpected changes 1`] = `
<div
  className="show-card"
>
  <img
    className="show-card-img"
    src="public/img/posters/undefined"
  />
  <div
    className="show-card-text"
  >
    <h3
      className="show-card-title"
    />
    <h4
      className="show-card-year"
    >
      (
      )
    </h4>
    <p
      className="show-card-description"
    />
  </div>
</div>
`;
